README.md

```diff
@@ -1,10 +1,376 @@
+
+### Install the dependencies
+
+```sh
+npm install @cloudflare/vite-plugin wrangler --save-dev
+```
+
+### Add the plugin to your Vite config
+
+```ts
+// vite.config.ts
+
+import { defineConfig } from 'vite';
+import { cloudflare } from '@cloudflare/vite-plugin';
+
+export default defineConfig({
+  plugins: [cloudflare()],
+});
+```
+
+### Create your Worker config file
+
+```toml
+# wrangler.toml
+
+name = "my-worker"
+main = "./src/index.ts"
+compatibility_date = "2024-12-05"
+```
+
+### Create your Worker entry file
+
+```ts
+// src/index.ts
+
+export default {
+  fetch() {
+    return new Response(`Running in ${navigator.userAgent}!`);
+  },
+};
+```
+
+## Tutorial
+
+In this tutorial, we're going to create a React SPA that can be deployed to Workers Assets.
+We'll then add an API Worker that can be accessed from the front-end code.
+We will develop, build and preview the application using Vite before finally deploying to Cloudflare.
+
+### Scaffold a Vite project
+
+Let's start by creating a React TypeScript project with Vite.
+
+```sh
+npm create vite@latest cloudflare-vite-tutorial -- --template react-ts
+```
+
+Open the `cloudflare-vite-tutorial` directory in your editor of choice.
+
+### Add the Cloudflare dependencies
+
+```sh
+npm install @cloudflare/vite-plugin wrangler --save-dev
+```
+
+### Add the plugin to your Vite config
+
+```ts
+// vite.config.ts
+
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import { cloudflare } from '@cloudflare/vite-plugin';
+
+export default defineConfig({
+  plugins: [react(), cloudflare()],
+});
+```
+
+### Create your Worker config file
+
+```toml
+# wrangler.toml
+
+assets = {}
+```
+
+The `directory` field is not used when configuring assets with Vite.
+The `directory` in the output configuration will automatically point to the client build output.
+
+> [!NOTE]
+> When using the Cloudflare Vite plugin, the Worker config (e.g. `wrangler.toml`) that you provide is the input configuration file.
+> A separate output `wrangler.json` file is created when you run `vite build`.
+> This output file is a snapshot of your configuration at the time of the build and is the configuration used for preview and deployment.
+
+### Configure not found handling
+
+If you run `vite dev`, you will see that your app is running.
+If you navigate to a different path, however, you will get a 404 response.
+This is because the not found handling is now being handled by the Cloudflare plugin rather than Vite and replicates the production behaviour that is specified in your configuration.
+To make all not found requests direct to the `index.html` file, set the value to `single-page-application` in your Worker config.
+
+```toml
+# wrangler.toml
+
+assets = { not_found_handling = "single-page-application" }
+```
+
+Now, if you navigate to a different path, you will still see your application.
+
+For a purely front-end application, you could now proceed to build and deploy your application.
```

Comment:
Perhaps put in a heading that is something like "Adding an API Worker" or similar.
The next heading being "Add the `@cloudflare/workers-types` dependency and configure TypeScript" felt jarring.

Reply:
I've changed the heading to 'Configure TypeScript for your Worker code'. Is that OK? Alternatively I could split the tutorial with two sub-headings and knock all the other headings down a level.